libsyclinterface/include/syclinterface/dpctl_sycl_extension_interface.h

```diff
@@ -53,4 +56,31 @@ void DPCTLWorkGroupMemory_Delete(__dpctl_take DPCTLSyclWorkGroupMemoryRef Ref);
 DPCTL_API
 bool DPCTLWorkGroupMemory_Available();
 
+typedef class RawKernelArgDataTy
+{
+public:
+    RawKernelArgDataTy(void *bytes, size_t count) : data(count)
+    {
+        std::memcpy(data.data(), bytes, count);
+    }
```

Comment:
Unfortunately, the use of copying and `cstring` also brought about another problem, namely, that this is now a class with a constructor under `extern "C"` linkage.

Builds fail but also, it won't be constructible from C with this constructor due to name-mangling issues.